src/blob_gc_job.cc

```diff
@@ -119,60 +119,62 @@ Status BlobGCJob::Run() {
     return s;
   }
 
-  std::string tmp;
-  for (const auto& f : blob_gc_->inputs()) {
-    if (!tmp.empty()) {
-      tmp.append(" ");
-    }
-    tmp.append(std::to_string(f->file_number()));
-  }
-
-  std::string tmp2;
-  for (const auto& f : blob_gc_->sampled_inputs()) {
-    if (!tmp2.empty()) {
-      tmp2.append(" ");
-    }
-    tmp2.append(std::to_string(f->file_number()));
-  }
-
-  ROCKS_LOG_BUFFER(log_buffer_, "[%s] Titan GC candidates[%s] selected[%s]",
-                   blob_gc_->column_family_handle()->GetName().c_str(),
-                   tmp.c_str(), tmp2.c_str());
-
-  if (blob_gc_->sampled_inputs().empty()) {
+  if (blob_gc_->gc_sampled_inputs().empty() && blob_gc_->fs_sampled_inputs().empty()) {
```

Comment:
If gc_sampled_inputs() is empty, we should skip the gc. If fs_sampled_inputs() is empty, we should skip the dighole. If gc_sampled_inputs() is empty and fs_sampled_inputs() is not, we can't skip the gc here. For the same reason we can't skip the dighole if gc_sampled_inputs() isn't empty.

Reply:
oh sorry I miss this obviously logic